src/index.js

```diff
@@ -226,74 +220,105 @@ const executeCommand = (command: string) =>
-            pRetry(run, {
-              onFailedAttempt: error => {
+  const defaultResult: InfluxDataResult = { stdout: '*' };
+  const targetDatabase = `${flags.newdb || flags.db}`;
+  const measurementsFields = useTargetMeasurements
+    ? [
+        executeCommand(`SHOW field keys ON ${targetDatabase}`, 'json'),
+        executeCommand(`SHOW tag keys ON ${targetDatabase}`, 'json'),
+      ]
+    : [Promise.resolve(defaultResult)];
+
+  return Promise.all(measurementsFields).then(([fields, tags]) => {
+    const measurements = useTargetMeasurements
+      ? parseResults(JSON.parse(fields.stdout))
+      : { '*': [] };
+    const tagKeys = useTargetMeasurements
+      ? parseResults(JSON.parse(tags.stdout))
+      : { '*': [] };
+
+    return Promise.all(
+      keys.map(key => {
+        const tempDatabase = `${flags.db}_${key}`;
+        return Promise.all(
+          Object.entries(measurements).map(([measurement, values]) => {
+            const run = async () => {
+              if (useTargetMeasurements) {
+                const tag = parseTags(tagKeys[measurement]);
+                const fieldKeys = parseFields(values, tagKeys[measurement]);
                 console.log(
-                  `Attempt ${
-                    error.attemptNumber
-                  } for ${tempDatabase} to ${targetDatabase} failed. There are ${
-                    error.attemptsLeft
-                  } attempts left.`,
+                  `Using target measurements:  ${tag.join(
+                    ',',
+                  )},${fieldKeys.join(',')}`,
                 );
-              },
-              retries: 5,
-            }),
-          );
-        }),
-      ).then(() => executeCommand(`DROP DATABASE ${tempDatabase}`));
-    }),
-  );
+                await executeCommand(
+                  `SELECT ${tag.join(',')},${fieldKeys.join(
+                    ',',
+                  )} INTO ${targetDatabase}..${measurement} FROM  ${tempDatabase}..${measurement} GROUP BY *`,
+                );
+              } else {
+                console.log('Using wild card for copying measurements');
+                await executeCommand(
+                  `SELECT * INTO ${targetDatabase}..:MEASUREMENT FROM  ${tempDatabase}../.*/ GROUP BY *`,
+                );
+              }
+
+              console.info(
+                `  INFO: Merged ${measurement ||
+                  `*`} ON ${tempDatabase} to ${targetDatabase}!`,
+              );
+              return Promise.resolve();
+            };
+            return limit(() =>
+              pRetry(run, {
+                onFailedAttempt: error => {
+                  console.log(
+                    `Attempt ${
+                      error.attemptNumber
+                    } for ${tempDatabase} to ${targetDatabase} failed. There are ${
+                      error.attemptsLeft
+                    } attempts left.`,
+                    `Consider using -useTargetMeasurements flag`,
```

Comment:
Should we add some heuristics to check if the error matches the common case we are running into?